allennlp/models/biaffine_dependency_parser.py

```diff
@@ -508,8 +502,25 @@ def _mst_decode(self,
     def _run_mst_decoding(batch_energy: torch.Tensor, lengths: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
         heads = []
         head_tags = []
-        for energy, length in zip(batch_energy.detach().cpu().numpy(), lengths):
-            head, head_tag = decode_mst(energy, length)
+        for energy, length in zip(batch_energy.detach().cpu(), lengths):
+            scores, tag_ids = energy.max(dim=0)
+            # Although we need to include the root node so that the MST includes it,
+            # we do not want any word to be the parent of the root node.
+            # Here, we enforce this by setting the scores for all word -> ROOT edges
+            # edges to be 0.
+            scores[0, :] = 0
+            # Decode the heads. Because we modify the scores to prevent
+            # adding in word -> ROOT edges, we need to find the labels ourselves.
+            head, _ = decode_mst(scores.numpy(), length, has_labels=False)
+
+            # We don't care what the head is for the root token, but by default it's
+            # returned as -1 - this messes up the indexing into the tag_ids tensor,
+            # so we'll just fix it to have a value of 0.
+            head[0] = 0
```

Comment:
Nit: I'd recommend `instance_heads` here, as this is still a list.  Same for `instance_head_tags`.